src/main/java/core/basesyntax/model/Action.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.model;
+
+public enum Action {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String code;
+
+    Action(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static Action fromCode(String code) {
+        for (Action action : values()) {
+            if (action.getCode().equals(code)) {
+                return action;
+            }
+        }
+        throw new IllegalArgumentException("Invalid operation code: " + code);
```

Comment:
Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead. However, in this particular case, the usage of a loop to find a matching enum is acceptable due to the nature of enums.